packages/core/src/ceramic.ts

```diff
@@ -287,32 +288,32 @@ class Ceramic implements CeramicApi {
 
   /**
    * Creates doctype from genesis record
+   * @param doctype - Document type
    * @param genesis - Genesis CID
    * @param opts - Initialization options
    */
-  async createDocumentFromGenesis<T extends Doctype>(genesis: any, opts: DocOpts = {}): Promise<T> {
-    const doc = await this._createDocFromGenesis(genesis, opts)
+  async createDocumentFromGenesis<T extends Doctype>(doctype: string, genesis: any, opts: DocOpts = {}): Promise<T> {
```

Comment:
It irks me a little that this, `findHandler`, and `createDocument` all need the doctype passed in as a string argument when they're already specialized on the Doctype, but my knowledge of typescript generics and inheritance still is still pretty basic and it didn't seem worth it for me to wade into that can of worms to try to figure out how to remove the string argument at this time.  Seems like it should be theoretically possible to get the type name from the type specialization though.

Reply:
I don't think you actually need to specify the generic when you call this method?

Reply:
interesting.  What type gets returned in that case?

Reply:
Types doesn't exist runtime, but compile time I guess it's just the generic `Doctype`.